volume/mountpoint/api.go

```diff
@@ -0,0 +1,154 @@
+package mountpoint
+
+import (
+	"os"
+
+	"github.com/docker/docker/api/types/mount"
+)
+
+// Type represents the type of a mount.
+type Type string
+
+// Type constants
+const (
+	// TypeBind is the type for mounting host dir
+	TypeBind Type = "bind"
```

Comment:
We could but eventually this enumeration will be bigger than the one in `api/types/mount`. For instance, layers and secrets could also be interposed on. This implementation currently doesn't support `tmpfs` so perhaps that variant should be removed from this type.